bootstrap_cfn/fab_tasks.py

```diff
@@ -483,35 +540,80 @@ def cfn_delete(force=False, pre_delete_callbacks=None):
+        print green("\nSTACK {0} DELETING...\n").format(stack_name)
+
+        # delete Alias record
+        elb_name = "{}-{}".format(elb, stack_id)
+        alias_record_object = r53_conn.get_full_record(get_zone_name(), get_zone_id(), elb_name, 'A')
+        if alias_record_object:
+            alias_record_value = [alias_record_object.alias_hosted_zone_id,
+                                  alias_record_object.alias_dns_name,
+                                  alias_record_object.alias_evaluate_target_health]
+            alias_record_name = "{}.{}".format(elb_name, get_zone_name())
+            r53_conn.delete_dns_record(get_zone_id(), alias_record_name, 'A', alias_record_value, is_alias=True)
+        # delete TXT record
+        txt_tag_record = get_tag_record_name(stack_tag)
+        txt_record_name = "{}.{}".format(txt_tag_record, get_zone_name())
+        txt_record_value = '"{}"'.format(r53_conn.get_record(
+                get_zone_name(), get_zone_id(), txt_tag_record, 'TXT'))
+        if txt_record_value:
+            r53_conn.delete_dns_record(get_zone_id(), txt_record_name, 'TXT', txt_record_value)
+
+        # Wait for stacks to delete
+        print 'Waiting for stack to delete.'
+        cfn.delete(stack_name)
+        if not env.blocking:
+            print 'Running in non blocking mode. Exiting.'
+            sys.exit(0)
+        tail(cfn, stack_name)
+        if cfn.stack_missing(stack_name):
+            print green("Stack successfully deleted")
+        else:
+            print red("Stack deletion was unsuccessful")
     else:
-        print red("Stack deletion was unsuccessfull")
+        # delete active dns records
+
+        stack_tag = 'active'
+        print green("\nDELETING ACTIVE DNS RECORDS...\n")
+
+        # delete 'A' record
+        main_record_name = "{}.{}".format(elb, get_zone_name())
+        stack_record_name = "{}-{}".format(elb, stack_id)
+        stack_record_object = r53_conn.get_full_record(get_zone_name(), get_zone_id(), stack_record_name, 'A')
+        main_record_object = r53_conn.get_full_record(get_zone_name(), get_zone_id(), elb, 'A')
+        main_record_value = [main_record_object.alias_hosted_zone_id,
+                             main_record_object.alias_dns_name,
+                             main_record_object.alias_evaluate_target_health]
+        if stack_record_object and stack_record_object.to_print() == main_record_object.to_print():
+            r53_conn.delete_dns_record(get_zone_id(), main_record_name, 'A', main_record_value, is_alias=True)
+
+        # delete 'TXT' record
+        tag_record_name = get_tag_record_name(stack_tag)
+        record_value = '"{}"'.format(r53_conn.get_record(
+            get_zone_name(), get_zone_id(), tag_record_name, 'TXT'))
+        record_name = '{}.{}'.format(tag_record_name, get_zone_name())
+        if stack_id and stack_id == record_value[1:-1]:
+            r53_conn.delete_dns_record(get_zone_id(), record_name, 'TXT', record_value)
```

Comment:
In this method you seem to be using get_zone_id() about 9 times. Does this get cached by the boto library? We shouldn't be repeating calls pointlessly as there is a risk to get throttled. 
Also I had created `get_cached_zone_id` that will do some caching when it looks up zone ids. Actually the whole get_zone_id() can be replicated by refactoring to this and passing an empty cache dict if you don't want caching: 
```python
def get_zone_id(r53_conn):
    return get_cached_zone_id(r53_conn, {}, get_zone_name())
```